### PR TITLE
Scrubber pressure check consistency

### PR DESCRIPTION
**Pull request by brndd**
[consistency] [bugfix]

Scrubbers won't suck air in if they're already at 50 ATM pressure or more, but this was only checked when they were siphoning, not when they were scrubbing. Now it's checked when they're scrubbing too.

:cl:
 * bugfix: Scrubbers now respect their pressure limit when scrubbing.

### CODE REVIEW

**Comment by ghost** [comment]
Feature.

**Comment by brndd** [comment]
> Feature.

Prove it, buddy.

**Comment by ghost** [comment]
Ok, I will empirically prove that this change in isolation is made from ignorance:

Passive vents and injectors have no pressure checks. What you should instead do is disable pressure checks on siphoning.

**Comment by brndd** [comment]
I went ahead and found the commit that originally added this pressure check, and it dates back to SVN oldcode from 2011 by rastaf0: https://github.com/vgstation-coders/vgstation13/commit/6b7801c4af182520d4381a373df1402c63bc46e2

Considering this, it's virtually guaranteed to be an oversight. If what you actually meant was "this is a bug, but I like/need/want this bug", then just say that instead.

**Comment by ghost** [comment]
I've made a PR which corrects the siphoning bug. https://github.com/vgstation-coders/vgstation13/pull/22882

**Comment by ShiftyRail** [comment]
Why should siphoning/scrubbing not work above a certain threeshold ?
What is the reasoning behind that ?

**Comment by ghost** [comment]
Consistency is good, but scrubbers being broken like that has been common knowledge for a long while without any issues.

Worth considering is that it'll kill big boys canisters, and will cause hot loops in TEGs to clog up at an even lower pressure with the scrubber+vent combo than they did with volumetric pumps _(5066.25 kPa vs 9000 kPa)_, though the vents will ease that up a tad. It'll also make cooling the waste line actually worthwhile, though that's not bad at all.

I think making it consistent has downsides and upsides, so I personally don't mind either way, but it's worth considering the _"features"_ it'll affect in the process.

**Comment by brndd** [comment]
> Why should siphoning/scrubbing not work above a certain threeshold ?
> What is the reasoning behind that ?

Presumably the pump in the scrubber unit isn't made of magic and can't operate beyond a certain pressure.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
@ShiftyRail Scrubbers are the only atmos equipment that allows you to infinipressure pipes like that and they are the end-all be-all atmos crutch to the point where consistently people make 1-tile vent+scrubber chambers to emulate a pump with no pressure limit

**Comment by ghost** [comment]
God I hope this actually gets merged.  I've been waiting forever for someone to make this PR.  Finally someone that understands that flooding should not be fixable and should be a round ender with no chance of recovery with the right sabotage applied.

**Comment by ghost** [comment]
Yes, it's wise to consider that with the default waste setup, this change means that rather than just the pump between the waste line and the vent line _"clogging"_ up once the vent line passes 9000 kPa, it'll actually clog up the scrubbers, too.

That might be preferable to scrubbers being able to infinitely pressurise pipenets, and it certainly gives an incentive to cool the vent line and cut the volumetric pump out, but consider the effect it'll have on rounds where there isn't anyone with a basic knowledge of atmospherics _(or a will)_ to fix it up.

**Comment by I-VAPE-VOX-CLOACA-EVERY-DAY-OF-MY-LIFE** [comment]
Scrubbers don't do SHIT against plasmafloods even in a best case scenario so no difference there

**Comment by ghost** [comment]
Obviously not, seeing as you're more likely to RCD the floor out or vent it to space some other way, but it's good to consider in the case of not-so-efficient floods _(e.g. most of them)_ or for all of the hot hot gas that'll get whipped away while someone's running over with an RCD _(as if someone won't just call the shuttle immediately, anyway)_.

Either way, it's a nice change, and we can always tweak the specifics after the fact if there're _unforeseen consequences._

**Comment by ghost** [comment]
It's more about nullifying distro and the waste loop altogether.  Not that they are that far apart, that's only one dinky manifold apart in a dark corner of the station given our magic pipes.  This makes that waaaaaaaay easier to completely bork both systems via heat clogging which was always my absolute favorite form of sabotage.  I'm thrilled at how much easier all of this will make atmos antag'ing and how it will take way more know-how and way more time to correct.  Its dumb how problems can be resolved in like 30-45 minutes and it will be way cooler to see them take a minimum of 2-3hours if at all.  A new age of atmos sabotage!

**Comment by FudgePucker** [comment]
@HarseTheef 
Are you for real?
Atmos shittery is the most EZ way to bork the station.
This will nerf fun canisters and buff fucking distro and waste.

I do not like change.
How will we provide methods of improving the amount of pressure you can acquire?
If you say cargo, science, or FUCKING TRADERS, you can FUCK RIGHT OFF.

**Comment by FudgePucker** [comment]
Also this will nerf round start set ups of SME on box, the scrubbers into the canister will eventually stop working.

**Comment by ghost** [comment]
To be fair, the default set up has always been terrible, but I get what you mean.

Either way, I think the implication is that they don't want us to be able to pressurise things above some 9000 kPa. It's fair, of course, because it is something of an exploit, but it almost feels like feature loss at the same time.

That aside, you could _"improve pressure"_ by foaming gas to 1k, filling a canister to capacity, and then heating it up again. That's... quite a bit more work than just slapping a scrubber and a vent down, and it's likely not worth it, but it's there.

**Comment by FudgePucker** [comment]
Boy having to do that shit would be fucking tedious as fuck though.
I can't wait to have to drag 100 canisters of air to refill breaches instead of just having one super filled up.
What a great fucking improvement.

**Comment by ghost** [comment]
Oh, yeah, don't get me wrong, it sounds like hell and I hate the prospect, but it's there.

**Comment by brndd** [comment]
>  I can't wait to have to drag 100 canisters of air to refill breaches instead of just having one super filled up.

How about, get this, fixing the distro line and letting the vents do their job? Not everything needs to be 🚀

**Comment by brndd** [comment]
@HarseTheef 
I'm sorry but you can still drain all the gas you need from waste by putting HE pipes and vents out in space (active vents have no pressure limitations when you turn pressure checks off, passive vents never have them but only vent their volume's worth of gas every tick), and replacing the volume pump with a valve.

**Comment by ghost** [comment]
Oh jeez, I forgot the part where the guy sabotaging atmospherics stops and let's you optimize the waste system and doesn't sabotage that too so you can foil his evil plot.  Extremely ideal thinking with someone actively sabotaging atmospherics.  Like I said, once distro and waste are connected, those sorts of big moves are another whole can of worms with ruining the station further/killing more with protagonist behavior!  Its not about what can be done ideally, its about this making it very easy to push things into spiraling out of control as an atmos antag, exactly the way it should be.  Seeing it as fixing only a flood is very one dimensional and will cause you to miss the beauty of just how much raw chaos this causes and atmos antagonistic goodiness this is.  Take into account that the absolute simplest move is connecting distro to waste in a far off corner that will never be found.  From that point on, everything that can be done will put you as responsible when it comes to fixing distro.  You can't freeze it to clear it out, that would kill ever more.  The heat is gonna clog both loops and this will fully grind all scrubbers to a halt everywhere, making simple flood vastly more effective given that the current biggest hang up is the effectiveness of scrubbers in halting the start of a flood in the absence of immediate needs to refill air.  It essentially will remove the need to even use a pressure check disabler (although its use will make things significantly worse so its always recommended).  This PR gives us the missing step that was always needed to truly cement the terror of a flood, and that is giving a very easy way to freeze the activity of the entire scrubber systems of the station while simultaneously improving the distribution of whatever evil concoction you have mixed without needed to micro manage everything via the AAC!  That along with completely destroying distro as well and removing the ability yo make those stupid refill canisters so that there will be no way to turn around and actually return the air!  God it is so perfect.  Spacing the oxygen miner will be strong again too now that destruction of the reserve air hold will truly remove any chance of refill!  Its so perfect that I have been flipping out.     Let's forget that though, i'm for this, not against it.  This needs to be speedmerged tbh!

**Comment by brndd** [comment]
Hooking distro to waste is unfortunately not a problem with an easy solution given the way pipenets currently work. However, I fail to see how scrubbers having pressure checks is going to make the problem meaningfully worse. If distro is connected to waste and then superheated, scrubbers are not going to help with that problem. Not much is going to help with that problem besides shutting off all vents to prevent rooms from heating up any further, venting the entire pipeline to space and finding where the two are connected.

As for the oxygen miner getting spaced, I'd suggest turning on the recycling loop. It's that green pipeline on box that you always wondered what it did.

If you have problems with pumps getting clogged, try using passive gates or gas pumps rather than volumetric pumps. They have no pressure limit on the input side, and passive gates don't even have output pressure limits (but will at most equalize the input and output).

**Comment by ghost** [comment]
> As for the oxygen miner getting spaced, I'd suggest turning on the recycling loop. It's that green pipeline on box that you always wondered what it did.

Always wondered, huh?  Gonna recycle the oxygen and nitrogen out of the distro/waste line to put it back into the distro/waste line, bud?  Or are you gonna recycle the nothing out of the distro/waste line because you dumped it so you can put nothing back into distro/waste?  Gonna throw in the temperature correcting loop and system to handle readjusting your mix for the air and handle a few tens of thousands of kpa with a single heater and cooler given that a combination of gas filter rates, bottlenecking of the system given how gas filters function, and dangerous temperatures have always been the biggest issues with the recycling system along with the futility of the system once the waste and distro loop are combined with a single manifold?

>  If distro is connected to waste and then superheated, scrubbers are not going to help with that problem

In the current system this does not half the scubbers upon the station, after this the scrubbers will cease to function throughout the station put the join loops filling, allowing the follow to continue with additional force everywhere with absolutely no progress outside of manual fixes to each affected location.  You underestimate just what the scrubberss actually do in the event of a minimal efffort flood.  

> If you have problems with pumps getting clogged, try using passive gates or gas pumps rather than volumetric pumps.

passive gates, huh?

You have a lot of the right idea on the fixes, but I don't think you truely grasp the time scale of what it would take to implement those solutions.  That's my favorite part, too.  Its going to be significantly easier to cause a lot more damage, with a lot less effort needed as a lot of the AAC implements are being handed mechanically now with these changes, and with a lot less solutions to the issues outside of a ton of manual fixes applied over a few hours as most are being removed.  Its gonna be awesome!

**Comment by brndd** [comment]
@HarseTheef 
> Always wondered, huh? Gonna recycle the oxygen and nitrogen out of the distro/waste lin [wordswordswords]

Like I said, combining distro and waste is a problem with few good solutions. It's nothing new, it's not unique to this PR, and it has next to nothing to do with scrubbers having pressure bounds. If it does happen, fixing it is obviously your first priority as an atmos technician. Assuming nothing else is wrong with the system (ie. no extra heat in the now combined distrowaste), a stopgap measure to keep the air going is to enable the recycling loop to keep removing CO2 and other undesired gases from the system while still keeping oxygen and nitrogen in.

>In the current system this does not half the scubbers upon the station, after this the scrubbers will cease to function throughout the station put the join loops filling, allowing the follow to continue with additional force everywhere with absolutely no progress outside of manual fixes to each affected location. You underestimate just what the scrubberss actually do in the event of a minimal efffort flood.

You're making very little sense right now. Are you describing a situation where distro is flooded with plasma *and* hooked to waste? Just turn off vents, make sure scrubbers are removing plasma, and start venting waste to space. If it's a "minimal effort flood" (ie. done using the machines found in atmos at roundstart), pressure is not a concern because all those machines share the pressure limit of scrubbers. If there's overpressure in the pipes, just vent them to space better.

>passive gates, huh?

Yes. You get them from the RPD and they're basically gas pumps that use no power and don't actually pump (and as a result of not pumping, have no upper external pressure bound). They're good if your waste is supercooled and you need to move gas from distro to waste without supercooling distro too. #22879 fixes gas pumps and passive gates so that they actually function as they should.

**Comment by strawberrytan** [comment]
I think waste actually being able to get full is a good change to try out

**Comment by brndd** [comment]
PS: there's still a way to make high-pressure canisters and such besides cooling gas and heating it if all my PRs pass. No, I won't tell you what it is. Discovery is part of the game etc.

**Comment by ghost** [comment]
> Assuming nothing else is wrong with the system (ie. no extra heat in the now combined distrowaste), a stopgap measure to keep the air going is to enable the recycling loop to keep removing CO2 and other undesired gases from the system while still keeping oxygen and nitrogen in.

I don't know how you can assume a room temperature waste loop -ever- with a straight face.  The recycling loop is an absolute newbie trap that will get you in a lot of trouble in the worst case scenario, and best case will just be a heavily clogged series of of filters.

> You're making very little sense right now. Are you describing a situation where distro is flooded with plasma and hooked to waste? Just turn off vents, make sure scrubbers are removing plasma, and start venting waste to space. If it's a "minimal effort flood" (ie. done using the machines found in atmos at roundstart), pressure is not a concern because all those machines share the pressure limit of scrubbers. If there's overpressure in the pipes, just vent them to space better.

I really don't know what to say.  Most floods always involve a distro connected to waste, however it is usually just lazily done in atmospherics itself.  It's not like that is even a bold assumption.  Also, you say turn off the vents, how do you intend to do that?  Like do you expect everyone to have a station wide AAC script specifically for that, or by hand via the master controls?  Pressure is absolutely always a concern, because even with a minimum effort flood, hot gases that return scrubbed from a fire dramatically increase the preasure in the waste loop.  Even the minimum effort gasses easily put distro-waste at its maximum capacity, especially given how full distro usually is.

> Yes. You get them from the RPD and they're basically gas pumps that use no power and don't actually pump (and as a result of not pumping, have no upper external pressure bound). They're good if your waste is supercooled and you need to move gas from distro to waste without supercooling distro too. #22879 fixes gas pumps and passive gates so that they actually function as they should.

Out of curiosity, can you tell me a situation where a passive gate would be ideal over a gas pump?  Also it is a really bad idea to have the balance of one PR dependent on the another active PR getting merged.  I like this PR a lot so you shouldn't do that or at least correct that so it isn't the case.

**Comment by brndd** [comment]
>Also, you say turn off the vents, how do you intend to do that?

#22869

>Also it is a really bad idea to have the balance of one PR dependent on the another active PR getting merged.

I specifically split this change off from the rest of the atmos PRs because I knew it would cause sore bottoms.

>Out of curiosity, can you tell me a situation where a passive gate would be ideal over a gas pump?

They work without power, which isn't a big deal (you can almost always expect to have power), but it's there. ~~They have no external pressure bound, while gas pumps have an external pressure bound of 4500 kPa.~~ wait what the fuck am I saying yes they do

**Comment by ghost** [comment]
Really cool, but the split does unfortunately make the balance of this PR dependent on that one getting merged, which makes it hard to approach and discuss this one until then.  Really good work, I hope all of them get pushed thru, its gonna be fantastic to get to use all of these new antag tools with most of the counters removed or heavily time restrained!

**Comment by strawberrytan** [comment]
Good thing you don't play anymore :^)

**Comment by ghost** [comment]
> They work without power, which isn't a big deal (you can almost always expect to have power), but it's there. 

So you changed them to fully function without power, or are they still bugged?

**Comment by brndd** [comment]
> So you changed them to fully function without power?

No, they've always functioned (or were supposed to function) without power. I'm changing them to be basically a one-way check valve now though, because as they currently are they really are pretty pointless and barely ever used. This means removing the pressure feature from them: now they'll stabilize output with input if input has higher pressure. It'll be in the vent sanity PR where the stuff related to (and necessary for) this is.

**Comment by ghost** [comment]
Well while you are at it, you should fix them being unable to be interacted with if they were placed prior to a room losing power, because currently they are not interact-able/do not function fully in a powerless room unless you wrench and unwrench the passive gate.  Such has been the case for many years.

**Comment by LordMyroc** [comment]
>How about, get this, fixing the distro line and letting the vents do their job? Not everything needs to be rocket

I am conflicted about this PR in general because I've always thought using scrubbers for infini-pressure is kinda gay, but I will say this: Any major station-wide issue that can't be fixed in 5-10 minutes or less will always, ALWAYS result in a shuttle call because no one outside of the people who actually know how to fix the problem have the patience to wait. With atmos sabotage being capable of gunking rounds hard enough as it is, think very carefully before removing ways to counteract it.

**Comment by brndd** [comment]
I don't see how scrubbers having no pressure bounds counteracts anything. It's primarily a consistency issue, and the main gameplay effect it will have is that waste now needs to be drained for scrubbers to keep working.

Like I said above, creating high-pressure canisters and such is still possible, just not by using scrubbers. It took me like 2 minutes to set up the rig to provide the screenshot above, and then 5-10 minutes to get the canister to the 200k pressure in the screenshot.

**Comment by ghost** [comment]
You haven't described what method you used so nobody can verify you're not telling tall tales. If you want to brag and keep secrets then there's a thread for shitposting.

**Comment by brndd** [comment]
Big boy canisters are hardly necessary for any mechanic to work as designed. They're essentially exploit territory. I don't see why I would be obligated to tell you how to exploit the game.

I also have no motive to lie about this. Prove your pipe autism is strong enough for your opinion to have merit on an issue that only affects extreme pipe autism and figure it out yourself. It's not difficult.

**Comment by ghost** [comment]
Your motive to lie is clear: you want to convince people that the one thing they disagree with when it comes to this change is still possible. So you post a pretty screenshot and make up a tale and go teehee I'm not actually going to tell you how.

I don't have to prove anything to you, you're making claims.

**Comment by brndd** [comment]
If people disagree with fixing an exploit because they want to keep using that exploit, I don't see why their opinion should be valued. Treat this PR as completely removing the only exploit that lets you make overpressurized canisters if you want to, I'm fine with that. The emojis, or a server poll should it come to that, will decide whether the exploit gets gone or not.

I have no love for overpressurized canisters. For one, they make items like the industrial vaporizer quite pointless. I don't think they should be removed entirely, I just think they should be harder to acquire than they currently are. Big rewards should require big effort, and eg. cooling and then heating gas is in my opinion sufficient effort.

It's entirely possible that portable pumps and scrubbers need to be looked at, because they really feel quite useless now. That'll probably end up happening sooner or later in my atmos fixing quest.

**Comment by Kammerjunk** [comment]
disregarding Harse's epic irony and sarcasm and dark humour, this is probably the important bit to address because everything else relates to it:
>If people disagree with fixing an exploit because they want to keep using that exploit, I don't see why their opinion should be valued.

if an eight-year-old exploit has become such a popular part of doing actual, proper atmos work (such as clearing a flood, which is what people I recognise as having done a lot of engineering, like Harse and Myroc, have brought up and which will still be as easy to cause if this gets merged) that most people don't realise it's an exploit, addressing it _as_ an exploit is dishonest. removing a bug, exploit, or oversight *because* it's a bug, exploit or oversight regardless of how it has been used since its introduction is not good (((game design))) and were it the industry standard, it would have removed a lot of wildly popular features (bunnyhopping, blast surfing). there is no reason "this is unintended" should be an end-all argument.

**Comment by ghost** [comment]
@brndd

You're the kinda guy that joins a quake server, uploads a fix to remove strafe-jumping and starts patting himself on the back about a good job because he removed an exploit and got upvotes from pubs.

You openly disregard opinions that disagree with you because they disagree with you. What are you even trying to achieve?

**Comment by brndd** [comment]
>a popular part of doing actual, proper atmos work (such as clearing a flood, which is what people I recognise as having done a lot of engineering, like Harse and Myroc, have brought up [...]

If there was an actual argument for why bypassing pressure checks is critical for clearing plasmafloods somewhere below all the badposting, I must've missed it. This should not have an effect under standard conditions where the waste loop is properly being emptied of gas. One of the big arguments above seemed to be about hooking distro to waste, and that has nothing to do with scrubbers or their pressure limits.

>removing a bug, exploit, or oversight because it's a bug, exploit or oversight regardless of how it has been used since its introduction is not good (((game design))) and were it the industry standard, it would have removed a lot of wildly popular features (bunnyhopping, blast surfing).

There's a big difference between bunnyhopping and being able to put 300000 kPa in a canister. There is nothing in the game that *requires* the kind of pressures that can only be achieved by this exploit, and it's mostly used as a way to bypass the (presumably) intentional difficulty involved in achieving pressures higher than the pressure bounds of most atmos machines.

Achieving gigapascals of pressure should be more difficult than wrenching down two pipes, a vent and a scrubber under an inflatable door. That way doing it would also feel more rewarding. It should *not* be part of "actual, proper atmos work" because it's not actually necessary for any actual, proper atmos work, and as a matter of fact generally lets you avoid doing said actual, proper atmos work because you can just squirt some air out of your canister of holding.

**Comment by brndd** [comment]
@codeungabunga 

>You openly disregard opinions that disagree with you because they disagree with you. What are you even trying to achieve?

Me disregarding your opinion just means that I'm not going to change the PR to match whatever it is you want it to match. It would be quite pointless for me to do so anyway considering you already made that PR yourself. My regard for your opinion doesn't affect your capability to press the thumbs down button.

**Comment by ReindH** [comment]
Did you test if scrubbers actually work after this nerf? A single fire can clog the waste line with 2000kPa hot gas for minutes, limiting the scrubbers can lead to the damage escalating.

**Comment by PJB3005** [comment]
Not sure whether my positive opinion of this PR is a good thing to some people, but anyways:

It's definitely gonna be interesting to see how the atmos meta will adapt, but still. The only thing this affects is people striving to get insane numbers. That's it. I do not see how this will even remotely affect regular station operation or waste loop management.

80% of this thread is just blatant strawmanning.

I don't think we should design the game based on the intentions of an oldcoder who missed a spot 8 years ago though. Personally I like this change, but we should still think critically here.

Also, if you actually cannot figure out a way to make a max pressure canister, you should re-evaluate whether you are actually experienced with engineering: 
image

This screenshot was taken with the change, no fuckery.

File: code/ATMOSPHERICS/components/unary/vent_scrubber.dm [hunk 1]
```diff
@@ -162,6 +162,8 @@
 			(scrub_N2O && environment[GAS_SLEEPING] > 0) ||\
 			(scrub_O2 && environment[GAS_OXYGEN] > 0) ||\
 			(scrub_N2 && environment[GAS_NITROGEN] > 0))
+			if (air_contents.return_pressure()>=50*ONE_ATMOSPHERE)
```
Comment (PJB3005) [inline code/ATMOSPHERICS/components/unary/vent_scrubber.dm]: Make `50*ONE_ATMOSPHERE` a define or something so you don't have to duplicate the value.

**Review by PJB3005 — CHANGES_REQUESTED** [review]

**Comment by ReindH** [comment]
PJB
The problem isn't really with the megacans, you can put 20000kPa into a can without scrubbers right now if you wish.
The real problem (in my opinion) is how any atmos misshap escalates without the scrubbers acting as a brake. Scrubbers already need considerable time to suck heated gasses out of rooms but this pull pretty much disables them.
I just don't want the shuttle called every time some greyshirt sets the library on fire.

**Comment by brndd** [comment]
> Scrubbers already need considerable time to suck heated gasses out of rooms but this pull pretty much disables them.

Just to be clear, this doesn't affect the ability of scrubbers to suck gas *out* of overpressurized rooms. The room could be ARBITRARILY_LARGE_NUMBER kPa and scrubbers would continue to remove gas from it. This just affects their ability to put gas into overpressurized pipes.

**Comment by ReindH** [comment]
But the PR said: Scrubbers won't suck air in if they're already at 50 ATM pressure or more.
Now you say they do. Does the gas get stuck in the scrubber then?

**Comment by brndd** [comment]
The pressure of the scrubber itself is equal to the pressure of the pipes it's attached to. If the pipe pressure (and therefore the pressure of the scrubber) is over 50 ATM (about 5066 kPa), the scrubber stops taking gas from the room and putting it in the pipe.

The pressure of the room itself (which is not the same as the scrubber's pressure) is irrelevant for the scrubber's functioning.

**Comment by ReindH** [comment]
I don't know how to break the news to you, but there is a correlation between sucking in heated gasses and making the waste loop hit the pressure limit.

**Comment by PJB3005** [comment]
Good thing Nanotrasen provided this handily connected vent to dump the waste loop into space for you then.

**Comment by brndd** [comment]
>I don't know how to break the news to you, but there is a correlation between sucking in heated gasses and making the waste loop hit the pressure limit.

What PJB said, basically. The waste loop has a very large volume. It will take a large-scale plasma fire to overpressurize it even with the default setup, and all it takes to fix that problem is to swap the volume pump in atmos to a digital valve and possibly add some extra cooling and layered vents outside.

**Comment by FudgePucker** [comment]
The industrial vaporizer is not pointless because canisters.
The industrial vaporizer is pointless because traders are literal trash who 95% of the time just smash into places to get loot.

I have had traders smash into engineering from space, you want to know how many times I have been offered the opportunity to buy an industrial vaporizer?
Zero.
Zero times.

**Comment by LordMyroc** [comment]
My personal experience with plasmafires beg to differ, I usually see the waste loop clog very quickly unless it was specifically upgraded with cooling prior to the flood. On the other hand, this was before the PR that nerfed fires by a whole lot and I haven't experienced many of those after (at least not where I've had an opportunity to look at the waste loop) so maybe that should be taken with a grain of salt.

I'm willing to see how this works in practice first in case this all turns out to be just doomsaying. I'm just skeptical of the dual mindsets of "fixing exploits simply because they're exploits" and "not every problem should be easy to fix". The former for the same reasons that have already been said that just because it's an exploit doesn't mean it's bad for the game, and the latter because plasmaflooding not only gunks the round very hard, but is considerably more difficult to clean up than it is to cause even before you take this change into consideration.

**Comment by PJB3005** [comment]
> I usually see the waste loop clog very quickly unless it was specifically upgraded with cooling prior to the flood.

This is completely reasonable IMO. The station shouldn't be perfect sabotage proof without setup.

**Comment by FudgePucker** [comment]
sabotage proofing the station is 1000 times harder than sabotaging it.
>LMAO I JUST PUT PLASMA INTO THE MIX TANK AND PUT IT INTO DISTRO, TIME FOR SHUTTLE CALL!

**Comment by PJB3005** [comment]
> sabotage proofing the station is 1000 times harder than sabotaging it.

This goes for everything and is a completely different problem.

**Comment by brndd** [comment]
>LMAO I PRESS ++ TWICE AND LOOSE THE GOOSE TIME FOR SHUTTLE CALL

>LMAO I WELDERBOMB THE SUPERMATTER TIME FOR SHUTTLE CALL

>LMAO I TOOK THE BOARD OUT OF THE CLONING MACHINE TIME FOR SHUTTLE CALL

the list goes on

**Comment by Kammerjunk** [comment]
regardless of how many methods there are of doing that, this is not isolated from the issue

File: code/ATMOSPHERICS/components/unary/vent_scrubber.dm (hunk 1 above)
Comment (brndd) [reply code/ATMOSPHERICS/components/unary/vent_scrubber.dm]: done

**Review by PJB3005 — APPROVED** [review]

File: code/ATMOSPHERICS/components/unary/vent_scrubber.dm (hunk 1 above)
Comment (Exxion) [reply code/ATMOSPHERICS/components/unary/vent_scrubber.dm]: Why not a var?

Comment (brndd) [reply code/ATMOSPHERICS/components/unary/vent_scrubber.dm]: These are defines in all the other files, but I guess there's no reason it couldn't be a variable. I kind of want to do some pipecode refactoring later so maybe I'll just leave var-izing it for then.

Comment (Exxion) [reply code/ATMOSPHERICS/components/unary/vent_scrubber.dm]: It definitely should be a var at some point but it doesn't have to be right now, so no problem

**Comment by Kurfursten** [comment]
ZAS forgive them

**Pull request merged**